public/app/core/hooks/useCleanup.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useRef } from 'react';
+import { useDispatch } from 'react-redux';
+import { cleanUpAction, StateSelector } from '../actions/cleanUp';
+
+export function useCleanup<T>(stateSelector: StateSelector<T>) {
+  const dispatch = useDispatch();
+  //bit of a hack to unburden user from having to wrap stateSelcetor in a useCallback. Otherwise cleanup would happen on every render
+  const selectorRef = useRef(stateSelector);
```

Comment:
hmm, not really sure what you want to do here. Looks like you want to dispatch the result from a selector when unmounting the component. Are you storing in in a ref so you don't have to add it to the useEffect dependency list?

Reply:
correct!
dep list is checked by reference. I want to avoid the scenario where user neglects to use `useCallback` and provides a new `stateSelector` on every render, thus triggering the effect and wiping redux state

Reply:
> Just had a comment, very hard to give a valuable review on this since I have very little insight in what's needed and not.

That is fine! Even a quick skim is much appreciated. 
Just unless there are serious concerns, please click approval button at the end. With tight deadline, I'd like to be merging things faster!